### PR TITLE
user API for model fitting plugin

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request extends on #1401 and provides user-API access to the model fitting plugin.

[Plugin User API docs](https://jdaviz--1658.org.readthedocs.build/en/1658/api/jdaviz.configs.default.plugins.model_fitting.model_fitting.ModelFitting.html#jdaviz.configs.default.plugins.model_fitting.model_fitting.ModelFitting)

**TODO**:
- [x] test and add test coverage
- [x] add PR to changelog if before next release, otherwise add new entry (waiting until PR is accepted otherwise to avoid constant merge conflicts)

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/default/plugins/model_fitting/model_fitting.py
Comment (kecnry) [inline jdaviz/configs/default/plugins/model_fitting/model_fitting.py:57]: this is a little ugly (that the class is called `AutoLabel` but being used for an equation, now that it is being exposed to the user).  I'm open to renaming the class, maybe something like `AutoTextField`?

File: jdaviz/configs/default/plugins/model_fitting/model_fitting.py
Comment (kecnry) [inline jdaviz/configs/default/plugins/model_fitting/model_fitting.py:527]: it might be nice to have this also take an API override for `results_label`, but that should probably be done across the board for all plugins at once and with shared code.

**Review by duytnguyendtn — APPROVED** [review]
Tested both in Cubeviz and Specviz; API methods are intuitive to follow along with the UI. Thanks!

File: jdaviz/configs/default/plugins/model_fitting/model_fitting.py
Comment (rosteen) [reply jdaviz/configs/default/plugins/model_fitting/model_fitting.py:61]: I don't think it's a big deal, but `AutoTextField` is a bit more general.

Comment (kecnry) [reply jdaviz/configs/default/plugins/model_fitting/model_fitting.py]: ok, I think now is the time to do it since we haven't made it public yet.  I just pushed that change, but if anyone can think of a better name before merge, I'm happy to swap it out again (it's just a find and replace)!

**Review by rosteen — APPROVED** [review]
Overall looks good, the only qualm I have is that I don't love the fitted models being returned immediately in the notebook when you run `calculate_fit()` - mainly because in the cube fitting case, the output is extremely long and I find it a little annoying. After the first time the user will know to throw it into a variable if they care though, so I don't think it's a dealbreaker, and I can't think of a good way that I'd prefer off the top of my head.

**Comment by kecnry** [comment]
Thanks for bringing that up @rosteen, I remember @camipacifici had some opinions on what should be returned as well, so let's hold off on merging until we make a final decision on that.

The `calculate_fit` method currently returns the fitted model object followed by the fitted spectrum.  As for the long output-cell, I'd say that could be addressed upstream by making the representation truncate the internal arrays, but we should still discuss if only one or both of these objects should be exposed.

**Pull request merged**
